### PR TITLE
docs: use testgrid for readme test status

### PR DESCRIPTION
**Pull request by tam7t**
**What this PR does / why we need it**: elevate test status visibility for non-PR tests

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +8,17 @@ Secrets Store CSI driver for Kubernetes secrets - Integrates secrets stores with
...
 
+## Test Status
+
+|Test|Status|
+|---|---|
+|periodic/image-scan|[![sig-auth-secrets-store-csi-driver-periodic/secrets-store-csi-driver-image-scan](https://testgrid.k8s.io/q/summary/sig-auth-secrets-store-csi-driver-periodic/secrets-store-csi-driver-image-scan/tests_status?style=svg)](https://testgrid.k8s.io/sig-auth-secrets-store-csi-driver-periodic#secrets-store-csi-driver-image-scan)|
+|periodic/azure-upgrade|[![sig-auth-secrets-store-csi-driver-periodic/secrets-store-csi-driver-upgrade-test-azure](https://testgrid.k8s.io/q/summary/sig-auth-secrets-store-csi-driver-periodic/secrets-store-csi-driver-upgrade-test-azure/tests_status?style=svg)](https://testgrid.k8s.io/sig-auth-secrets-store-csi-driver-periodic#secrets-store-csi-driver-upgrade-test-azure)|
+|postsubmit/aws|[![sig-auth-secrets-store-csi-driver-postsubmit/secrets-store-csi-driver-e2e-aws-postsubmit](https://testgrid.k8s.io/q/summary/sig-auth-secrets-store-csi-driver-postsubmit/secrets-store-csi-driver-e2e-aws-postsubmit/tests_status?style=svg)](https://testgrid.k8s.io/sig-auth-secrets-store-csi-driver-postsubmit#secrets-store-csi-driver-e2e-aws-postsubmit)|
```
Comment (nilekhc) [inline README.md:17]: Looks like svg is missing https://testgrid.k8s.io/q/summary/sig-auth-secrets-store-csi-driver-postsubmit/secrets-store-csi-driver-e2e-aws-postsubmit/tests_status?style=svg

Is this something we can fix?

Comment (tam7t) [reply README.md:17]: Yeah its just missing from the postsubmit dashboard, adding with https://github.com/kubernetes/test-infra/pull/22910

**Review by nilekhc — APPROVED** [review]
lgtm

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**